### PR TITLE
discord-development: init at {0.0.216,0.0.8778}

### PR DESCRIPTION
**Pull request by jopejoe1**
###### Description of changes
Updates  #179212 to latest version and resolve merge conflicts.

Closes  #179212

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/discord/default.nix [hunk 1]
```diff
@@ -49,7 +59,7 @@ let
     downloadPage = "https://discordapp.com/download";
     sourceProvenance = with sourceTypes; [ binaryNativeCode ];
     license = licenses.unfree;
-    maintainers = with maintainers; [ MP2E artturin infinidoge ];
+    maintainers = with maintainers; [ MP2E artturin infinidoge jopejoe1 ];
```
Comment (Infinidoge) [inline pkgs/applications/networking/instant-messengers/discord/default.nix:62]: This should be separated into its own commit

Comment (jopejoe1) [reply pkgs/applications/networking/instant-messengers/discord/default.nix:62]: done

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -39592,6 +39592,12 @@ with pkgs;
     branch = "canary";
   };
 
+  discord-development = import ../applications/networking/instant-messengers/discord {
+    inherit lib stdenv;
+    inherit (pkgs) callPackage fetchurl;
+    branch = "development";
+  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:39599]:
```suggestion
  discord-development = callPackage ../applications/networking/instant-messengers/discord {
    branch = "development";
  };
```

Comment (jopejoe1) [reply pkgs/top-level/all-packages.nix:39599]: Changing it to a callPackage would break overriding acording to https://github.com/NixOS/nixpkgs/pull/179212#discussion_r914086385

Comment (Artturin) [reply pkgs/top-level/all-packages.nix:39599]: https://github.com/NixOS/nixpkgs/pull/222950 btw

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:39599]: Then this needs a comment otherwise the next person is asking themselves possible the same thing as I did

Comment (jopejoe1) [reply pkgs/top-level/all-packages.nix]: Added a comment on top of all the discord packages

**Comment by Artturin** [comment]
Please do the backport
